docs/book/getting-started/deploying-zenml/helm.md

```diff
@@ -1,49 +1,155 @@
+recommended that you create a dedicated database user for the ZenML server and
+that you restrict its privileges to only access the database that will be used
+by ZenML. Enforcing secure SSL connections for the user/database is also
+recommended. See the [MySQL documentation](https://dev.mysql.com/doc/refman/5.7/en/access-control.html)
+for more information on how to set up users and privileges.
+- the name of the database that will be used by ZenML. The database does not
+have to exist prior to the deployment (ZenML will create it on first start).
+However, you need to create the database if you follow the best practice of
+restricting database user privileges to only access it.
+- if you plan on using SSL to secure the client database connection, you may
+also need to prepare additional SSL certificates and keys:
+
+  - the TLS CA certificate that was used to sign the server TLS certificate, if
+    you're using a self-signed certificate or signed by a custom certificate
+    authority that is not already trusted by default by most operating systems. 
+  - the TLS client certificate and key. Only needed if you decided to use
+    client certificates for your DB connection (some managed DB services support
+    this, CloudSQL is an example).
+
+### Optional cluster services
+
+It is common practice to install additional infrastructure related services in
+a Kubernetes cluster to support the deployment and long-term management of
+applications. For example:
+
+- an Ingress service like [nginx-ingress](https://kubernetes.github.io/ingress-nginx/deploy/)
+is recommended if you want to expose HTTP services to the internet. An Ingress
+is required if you want to use secure HTTPs for your ZenML deployment. The
+alternative is to use a LoadBalancer service to expose the ZenML service using
+plain HTTP, but this is not recommended for production.
+- [cert-manager](https://cert-manager.io/docs/installation/) is
+recommended if you want to generate and manage TLS certificates for your ZenML
+deployment. It can be used to automatically provision TLS certificates from a
+certificate authority (CA) of your choice, such as [Let's Encrypt](https://letsencrypt.org/).
+As an alternative, the ZenML Helm chart can be configured to auto-generate
+self-signed or you can generate the certificates yourself and provide them to
+the Helm chart, but this makes it more difficult to manage the certificates
+and you need to manually renew them when they expire.
+
+## ZenML Helm Installation
+
+### Configure the Helm chart
+
+To customize the Helm chart for your deployment, you should create a copy of
+the `values.yaml` file located in the `src/zenml/zen_server/deploy/helm` folder
+(let’s call this `custom-values.yaml`). You’ll use this as a template to
+customize your configuration. Any values that you don’t override you should
+simply remove from your `custom-values.yaml` file to keep it clean and
+compatible with future helm chart releases.
```

Comment:
```suggestion
compatible with future Helm chart releases.
```
In general, there's a mix of `helm` and `Helm` used in this file. Probably should standardise.